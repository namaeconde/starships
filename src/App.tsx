import { ThemeProvider } from "@mui/material/styles";
import theme, { color } from "./theme";
import CssBaseline from '@mui/material/CssBaseline';
import Box from "@mui/material/Box";
import HomePage from "./pages/HomePage";

function App() {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <Box
        display="flex"
        minHeight="100vh"
        bgcolor={ color.black_pearl }
      >
        <HomePage />
      </Box>
    </ThemeProvider>
  );
}

export default App;
