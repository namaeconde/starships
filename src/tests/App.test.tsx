import React from 'react';
import { render, cleanup } from '@testing-library/react'
import App from '../App';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const initialState = { favoriteList: { value: [] } }
const mockStore = configureStore()
let store

afterEach(cleanup);

test('App renders', () => {
  store = mockStore(initialState);
  const { asFragment }= render(<Provider store={store}><App /></Provider>,);
  expect(asFragment()).toMatchSnapshot();
});
