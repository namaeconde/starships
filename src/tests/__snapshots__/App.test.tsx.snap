// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1ftmrw2"
  >
    <div
      class="MuiBox-root css-8atqhb"
    >
      <div
        class="MuiBox-root css-l10m55"
      >
        <div
          class="MuiBox-root css-175rvmj"
        >
          <a
            href="/"
          >
            <div
              class="MuiAvatar-root MuiAvatar-square css-56sua1-MuiAvatar-root"
            >
              <img
                alt="Starship"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="/assets/home_logo.png"
              />
            </div>
          </a>
        </div>
        <div
          class="makeStyles-container-1 MuiBox-root css-vq34d8"
        >
          <a
            class="button muted-button"
            href="/favorites"
            style="text-decoration: none;"
          >
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root makeStyles-sidebarButton-3 css-ii4oqr-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              View favorites
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
        <div
          class="makeStyles-main-2 MuiBox-root css-1xkyxdb"
        >
          <div
            class="MuiBox-root css-uj0oyz"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-1ylcvda-MuiTypography-root"
            >
              Starship List
            </p>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined MuiBox-root css-1vtca5r-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiBox-root css-es3gb9-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-heightAuto css-112ph30-MuiSkeleton-root"
                style="width: 100%;"
              >
                <div
                  style="padding-top: 57%;"
                />
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiBox-root css-es3gb9-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-heightAuto css-112ph30-MuiSkeleton-root"
                style="width: 100%;"
              >
                <div
                  style="padding-top: 57%;"
                />
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiBox-root css-es3gb9-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-heightAuto css-112ph30-MuiSkeleton-root"
                style="width: 100%;"
              >
                <div
                  style="padding-top: 57%;"
                />
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiBox-root css-es3gb9-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-heightAuto css-112ph30-MuiSkeleton-root"
                style="width: 100%;"
              >
                <div
                  style="padding-top: 57%;"
                />
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiBox-root css-es3gb9-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-heightAuto css-112ph30-MuiSkeleton-root"
                style="width: 100%;"
              >
                <div
                  style="padding-top: 57%;"
                />
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiBox-root css-es3gb9-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-heightAuto css-112ph30-MuiSkeleton-root"
                style="width: 100%;"
              >
                <div
                  style="padding-top: 57%;"
                />
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiBox-root css-es3gb9-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-heightAuto css-112ph30-MuiSkeleton-root"
                style="width: 100%;"
              >
                <div
                  style="padding-top: 57%;"
                />
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiBox-root css-es3gb9-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-heightAuto css-112ph30-MuiSkeleton-root"
                style="width: 100%;"
              >
                <div
                  style="padding-top: 57%;"
                />
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiBox-root css-es3gb9-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-heightAuto css-112ph30-MuiSkeleton-root"
                style="width: 100%;"
              >
                <div
                  style="padding-top: 57%;"
                />
              </span>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiBox-root css-es3gb9-MuiGrid-root"
            >
              <span
                class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse MuiSkeleton-withChildren MuiSkeleton-heightAuto css-112ph30-MuiSkeleton-root"
                style="width: 100%;"
              >
                <div
                  style="padding-top: 57%;"
                />
              </span>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-container-1 MuiBox-root css-1icjmr2"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
